packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -151,15 +151,23 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
 
         const updatedMaterials =
             await threeGLTF.correlatedSceneGraph.loadVariant(variantName!);
-        const {gltf, gltfElementMap} = threeGLTF.correlatedSceneGraph;
-
-        for (const index of updatedMaterials) {
-          const material = gltf.materials![index];
-          this[$model]!.materials[index] = new Material(
-              this[$getOnUpdateMethod](),
-              gltf,
-              material,
-              gltfElementMap.get(material) as Set<MeshStandardMaterial>);
+        const {gltf, gltfVariantMap} = threeGLTF.correlatedSceneGraph;
+
+        if (gltfVariantMap !== null) {
+          for (const index of updatedMaterials) {
+            let variant = this[$model]!.variants.get(index);
+            if (variant === null || variant === undefined) {
```

Comment:
I don't think the `continue` works because in the case where `variant == null` there are still other duties to fulfill that must happen after this check which would get skipped if it were to continue, the null check is to just to create the variant on the fly/as-needed and to prevent it from being created more than once. Apologies if I'm misunderstanding you comment, sometimes I get tunnel vision after I've written something one way.

Reply:
This is moot with latest changes